tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/PublishCoreSetupBinaries.cs

```diff
@@ -40,18 +42,9 @@ public override bool Execute()
                 string binaryFileName = Path.GetFileName(binaryFullPath);
                 string version = Regex.Match(binaryFileName, VersionMatchRegex).Groups["semver"].Value;
 
-                // workaround the RID being included for now - regex needs to be reworked for stable versions
-                if (version.EndsWith("-linux"))
-                {
-                    version = version.Substring(0, version.Length - "-linux".Length);
-                }
-                if (version.EndsWith("-osx"))
-                {
-                    version = version.Substring(0, version.Length - "-osx".Length);
-                }
-                if (version.Contains("-ubuntu"))
+                foreach (var ba in BadAtoms)
```

Comment:
Order matters, is that ok? Should we loop until no more atoms exist, in case we run into weird subsequences later?

Reply:
Good point, added the loop.